### DISCUSSION TITLE
It requires more VRAM when you give bigger beam size is that expected?

### DISCUSSION THREAD

**Discussion by FurkanGozukara** [openai/whisper#396] [Q&A] [answered]
I have found out that bigger beam (10) size increases accuracy

For example by default it spells as GPX, however I am saying JPEG and it correctly spells with beam size 10 : https://www.youtube.com/watch?v=_nKwisL8dTs

However, with bigger beam size (20), i am starting to get not enough memory error on my 12 GB RTX 3060

Is this expected and why happening?

Moreover, what other hyper parameters did you find that increases accuracy?

**Comment by shervinemami** [comment, marked as answer]
Yes this is expected, they mention it in some of these GitHub discussions that you can fit in smaller RAM by turning off beam_size and best_of (so that you'll be using the Greedy decoder instead of BeamSearch Decoder). Also, using FP16 will use less RAM than FP32. I posted some GDDR measurements for small beam search values in  https://github.com/openai/whisper/discussions/391 where you can see that `beam_size=7` is slightly larger memory requirement than the others, but not by much.

**Reply by FurkanGozukara** [reply]
So with which settings do you think we could obtain higher accuracy? Bigger beam size what else?

**Reply by shervinemami** [reply]
The whisper developers mention that the best parameters will vary for different conditions, but they generally see slightly better accuracy by using larger models, and more beam searches. The other options such as temperature and best_of and patience are intended for stability not accuracy, since whisper often ends up in repeated behaviour, so the most accurate settings in some cases might cause instability in other cases. So you'll want to read the original paper and do some experimenting on your data to see which combo of settings works well for you.

(The end of the paper has make comparisons of settings and models with different datasets. Where you can see that for example, even the large model is sometimes less accurate than smaller models for some datasets! But in general, larger models and higher beam search count are helpful)

**Reply by FurkanGozukara** [reply]
ty for reply. yes with beam size 15 it started to fail transcribe. printing out ... now i am testing with --compression_ratio_threshold 3.0   --logprob_threshold -2.0 and beam size 15

**Reply by FurkanGozukara** [reply]
@shervinemami it produced a lot of ... when i increased beam size to 15 . any ideas why? I even tested compression_ratio_threshold and logprob_threshold same

**Reply by shervinemami** [reply]
Yes @FurkanGozukara I also get "..." inserted extremely often, since I mostly talk on short bursts, whisper inserts these triple dots around 1 or 2 times per sentence! I'm about to submit a pull request that disables it, because I don't expect any users will want these ... automatically inserted. I'll submit the PR today, it might not be merged but at least you'll be able to grab it locally to your system.

**Reply by FurkanGozukara** [reply]
@shervinemami but it puts them into the places where I talk :) When I use default settings I don't get them. Bigger beam size caused that a lot why do you think?

**Reply by shervinemami** [reply]
Whisper isn't a predictable recognition system, many things can have large impact on the results. I've seen "..." inserted lots when using some models and not others, or with some prompt values and not others, and some combinations of Decoder options and not others. Either way, I don't want any "..."! So I've created a Pull Request to stop it happening: https://github.com/openai/whisper/pull/399

**Reply by dgoryeo** [reply]
Hi @FurkanGozukara , I came across your old post on experiemnting with beam size. Did you by any chance arrive to any good balance between bean size, compression ratio, and logprobe?

**Reply by FurkanGozukara** [reply]
yes i am suing these settings with V1

string srCommand = @$"cmd /c whisper ""{srExtractMp3Name}"" --model large-v1 --language en --initial_prompt ""Welcome our Youtube channel."" --best_of 10 --beam_size 10 --output_dir ""{srFileDirectory}""";

**Reply by dgoryeo** [reply]
Thanks @FurkanGozukara . Am I right to understand that `best_of` parameter would be ignored because the temperature is zero and `beam search` active? Admittedly I've had difficulty to understand how `beam_size` and `best_of` play together.

**Reply by FurkanGozukara** [reply]
i am not sure could be. i did some testing and this yields best for my speech. but of course not enough. after this i do punctuation correction with AI and then turn back that into the VTT

**Reply by dgoryeo** [reply]
Thanks @FurkanGozukara.
